src/UCTNode.cpp

```diff
@@ -112,53 +111,55 @@ bool UCTNode::create_children(std::atomic<int> & nodecount,
         }
     }
 
-    link_nodelist(nodecount, nodelist, net_eval);
+    link_nodelist(nodelist, net_eval);
     return true;
 }
 
-void UCTNode::link_nodelist(std::atomic<int> & nodecount,
-                            std::vector<Network::scored_node> & nodelist,
+void UCTNode::link_nodelist(std::vector<Network::scored_node> & nodelist,
                             float init_eval) {
     if (nodelist.empty()) {
         return;
     }
 
     // Use best to worst order, so highest go first
     std::stable_sort(rbegin(nodelist), rend(nodelist));
+    // Reverse list so that emplace then expand orders by score
+    std::reverse(begin(nodelist), end(nodelist));
 
     LOCK(get_mutex(), lock);
 
-    m_children.reserve(nodelist.size());
+    m_child_scores.reserve(nodelist.size());
     for (const auto& node : nodelist) {
-        m_children.emplace_back(
-            std::make_unique<UCTNode>(node.second, node.first, init_eval)
-        );
+        m_child_scores.emplace_back(node.second, node.first);
     }
 
-    nodecount += m_children.size();
+    m_child_init_eval = init_eval;
```

Comment:
Rather than add m_child_init_eval, rebase on #888 to save this complexity.

Reply:
@sethtroisi Friendly ping for removing m_child_init_eval now that m_net_eval is available.